OpenRA.Mods.Common/Scripting/Properties/DockClientManagerProperties.cs

```diff
@@ -0,0 +1,88 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using Eluant;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Scripting;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptPropertyGroup("Ability")]
+	public class DockClientManagerProperties : ScriptActorProperties, Requires<DockClientBaseInfo>
+	{
+		readonly DockClientManager manager;
+
+		public DockClientManagerProperties(ScriptContext context, Actor self)
+			: base(context, self)
+		{
+			manager = self.Trait<DockClientManager>();
+		}
+
+		[ScriptActorPropertyActivity]
+		[Desc("Can we dock to this actor?")]
+		public bool CanDockAt(Actor target, bool forceEnter = false, bool ignoreOccupancy = false)
+		{
+			if (target == null)
+				throw new LuaException("Target actor is null.");
+
+			return manager.CanDockAt(target, forceEnter, ignoreOccupancy);
+		}
+
+		[ScriptActorPropertyActivity]
+		[Desc("Dock to the actor. Returns true if docking activity was successfully queued.")]
+		public bool Dock(Actor target, bool forceEnter = false, bool ignoreOccupancy = true)
```

Comment:
Non of our other methods return `true`/`false` for queuing activities afaik, so this shouldn't either.

Reply:
the alternative is to crash in all cases and force people to run `CanDockAt`

Reply:
Why crash? It would simply return and do nothing - the same as if the activity fails. That is why we don't return bools, since we can't tell the user if it will succeed or not. There should be a `Trigger.OnDock` (and related triggers) that can be used to determine if docking happened.

Reply:
Sure